### PR TITLE
Validate name field matches current spec's conventions

### PR DESCRIPTION
**Pull request by deruf**
Validation of schema names on the server side (not in the query document)

### CODE REVIEW

**Comment by dminkovsky** [comment]
Thanks for this. More schema definition validation is necessary and this is a good step in that direction!

**Comment by deruf** [comment]
Please don't accept this yet, I found a problem and have some additional validation to add.

**Comment by dminkovsky** [comment]
Sounds good. Thank you!

**Comment by dminkovsky** [comment]
Just dropping a few related notes:

* #227 can be closed after this is merged
* #215 is also good for follow up

**Comment by bbakerman** [comment]
@deruf - do you have follow up commits you want?  This seems worthwhile as it is?

File: src/main/java/graphql/Assert.java [hunk 1]
```diff
@@ -15,4 +15,18 @@
         return c;
     }
 
+    private static final String invalidNameErrorMessage = "Name must be non-null, non-empty and match [_A-Za-z][_0-9A-Za-z]*";
```
Comment (bbakerman) [inline src/main/java/graphql/Assert.java]: https://facebook.github.io/graphql/#sec-Names

Nice!

**Comment by bbakerman** [comment]
I am going to merge this PR and follow fixes can come in another one

**Pull request merged**